pvlib/atmosphere.py

```diff
@@ -683,3 +683,219 @@ def angstrom_alpha(aod1, lambda1, aod2, lambda2):
+        is set to min_pw to avoid model divergence. [cm]
+
+    max_pw : float, default 4
+        maximum atmospheric precipitable water. Any pw value higher than max_pw
+        is set to NaN to avoid model divergence. [cm]
+
+    module_type : None or string, default None
+        a string specifying a cell type. Values of 'cdte', 'monosi', 'cigs',
+        'multisi','asi' and 'pervovskite'. If provided,
+        module_type selects default coefficients for the following modules:
+
+            * 'cdte' - anonymous CdTe module.
+            * 'monosi', - anonymous sc-si module.
+            * 'multisi', - anonymous mc-si- module.
+            * 'cigs' - anonymous copper indium gallium selenide module.
+            * 'asi' - anonymous amorphous silicon module.
+            * 'perovskite' - anonymous pervoskite module.
+
+
+    coefficients : None or array-like, default None
+        the coefficients employed have been obtained with experimental
+        data in the city of Jaén, Spain. It is pending to verify if such
+        coefficients vary in places with extreme climates where AOD and
+        pw values are frequently high.
+
+    Returns
+    -------
+    modifier: array-like
+        spectral mismatch factor (unitless) which is can be multiplied
+        with broadband irradiance reaching a module's cells to estimate
+        effective irradiance, i.e., the irradiance that is converted to
+        electrical current.
+
+    References
+    ----------
+    .. [1] Gueymard, Christian. SMARTS2: a simple model of the
+        atmospheric radiative transfer of sunshine: algorithms
+        and performance assessment. Cocoa, FL:
+        Florida Solar Energy Center, 1995.
+    .. [2] Theristis, M., Fernández, E., Almonacid, F., and
+            Pérez-Higueras, Pedro. "Spectral Corrections Based
+            on Air Mass, Aerosol Optical Depth and Precipitable
+            Water for CPV Performance Modeling.
+            "IEEE Journal of Photovoltaics 2016, 6(6), 1598-1604.
+            https://doi.org/10.1109/jphotov.2016.2606702
+    .. [3] Caballero, J.A., Fernández, E., Theristis, M.,
+        Almonacid, F., and Nofuentes, G. "Spectral Corrections Based on
+        Air Mass, Aerosol Optical Depth and Precipitable Water
+        for PV Performance Modeling.
+        " IEEE Journal of Photovoltaics 2018, 8(2), 552-558.
+        https://doi.org/10.1109/jphotov.2017.2787019
+
+        """
+
+    # --- Screen Input Data ---
+
+    # *** ama ***
+    # Replace Extremely High ama with ama 10 to prevent model divergence
+    # ama > 10 will only occur very close to sunset
+    if np.max(airmass_absolute) > 10:
+        airmass_absolute = np.minimum(airmass_absolute, 10)
+
+    # Warn user about ama data that is exceptionally low
+
+    if np.min(airmass_absolute) < 0.58:
+        warn('Exceptionally low air mass: ' +
+             'model not intended for extra-terrestrial use')
```

Comment:
Question for the pvlib maintainers: `first_solar_spectral_correction` copied these warnings from the MATLAB implementation, but it seems a little out of place in pvlib-python in 2021.  Worth keeping?

Reply:
I don't think so.